uint8 mode
int16 x
int16 y
int16 z